kusto/kcsb.go

```diff
@@ -265,50 +257,68 @@ func (kcsb *ConnectionStringBuilder) WithSystemManagedIdentity() *ConnectionStri
 	return kcsb
 }
 
-func (kcsb *ConnectionStringBuilder) WithInteractiveLogin(clientID string, authorityID string, redirectURL string) *ConnectionStringBuilder {
+/*
+Creates a Kusto Connection string builder that will authenticate by launching the system default browser
+to interactively authenticate a user, and obtain an access token
+*/
+func (kcsb *ConnectionStringBuilder) WithInteractiveLogin(authorityID string) *ConnectionStringBuilder {
 	requireNonEmpty(dataSource, kcsb.DataSource)
 	kcsb.resetConnectionString()
-	if !isEmpty(clientID) {
-		kcsb.ApplicationClientId = clientID
-	}
 	if !isEmpty(authorityID) {
 		kcsb.AuthorityId = authorityID
 	}
-	if !isEmpty(redirectURL) {
-		kcsb.RedirectURL = redirectURL
-	}
 	kcsb.InteractiveLogin = true
 	return kcsb
 }
 
-func (kcsb *ConnectionStringBuilder) AttachClientOptions(options *azcore.ClientOptions) *ConnectionStringBuilder {
+/*
+Assigns ClientOptions to string builder that contains configuration settings like Logging and Retry configs for a client's pipeline.
+Read more at https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore@v1.2.0/policy#ClientOptions
+*/
+func (kcsb *ConnectionStringBuilder) AttachPolicyClientOptions(options *azcore.ClientOptions) *ConnectionStringBuilder {
 	requireNonEmpty(dataSource, kcsb.DataSource)
 	if options == nil {
 		kcsb.ClientOptions = options
 	}
 	return kcsb
 }
 
+// Create Kusto Conntection String that will be used for no auth mode. No auth token will be used for making requests.
+func (kcsb *ConnectionStringBuilder) WithNoAtuhentication() *ConnectionStringBuilder {
```

Comment:
No need for this API.
NoAuth mode should be the default behavior if no API is called.
DefaultAzureCredential should be an explicit choice, i.e. there should be an API named WithDefaultAzureCredential()